feature-libs/product-configurator/rulebased/components/overview-menu/configurator-overview-menu.component.ts

```diff
@@ -13,35 +21,156 @@ import { ConfiguratorStorefrontUtilsService } from '../service/configurator-stor
+  }
+
+  @HostListener('window:scroll', ['$event'])
+  onScroll(): void {
+    this.menuItem = this.getMenuItemToHighlight();
+    this.highlight(this.menuItem);
+    this.height = this.getHeight();
+    this.syncScroll(this.menuItem);
+  }
+
+  @HostListener('window:resize', ['$event'])
+  onResize() {
+    this.height = this.getHeight();
+    this.syncScroll(this.menuItem);
+  }
+
+  protected getHeight(): string {
+    return (
+      this.configuratorStorefrontUtilsService.getSpareViewportHeight() + 'px'
+    );
+  }
+
+  protected getMenuItemToHighlight(): HTMLElement | undefined {
+    let menuItem: HTMLElement | undefined;
+    const groups = this.configuratorStorefrontUtilsService.getElements(
+      this.CX_GROUPS
+    );
+    const verticallyScrolledPixels =
+      this.configuratorStorefrontUtilsService.getVerticallyScrolledPixels();
+
+    groups?.forEach((group) => {
+      if (
+        verticallyScrolledPixels &&
+        verticallyScrolledPixels >= group.offsetTop
+      ) {
+        const id = group.id.replace(this.OV_GROUP, this.OV_MENU_ITEM);
+        if (id) {
+          const querySelector = '#' + id;
+          menuItem =
+            this.configuratorStorefrontUtilsService.getElement(querySelector);
+        }
+      }
+    });
+    return menuItem;
+  }
+
+  protected highlight(elementToHighlight: HTMLElement | undefined): void {
+    if (elementToHighlight) {
+      const menuItems = this.configuratorStorefrontUtilsService.getElements(
+        this.CX_MENU_ITEM_BUTTONS
+      );
+      menuItems?.forEach((menuItem) => {
+        menuItem.classList.remove(this.ACTIVE_CLASS);
+        if (menuItem.id === elementToHighlight.id) {
+          elementToHighlight.classList.add(this.ACTIVE_CLASS);
+        }
+      });
+    }
+  }
+
+  protected syncScroll(element: HTMLElement | undefined): void {
+    if (
+      element &&
+      this.configuratorStorefrontUtilsService.hasScrollbar(
+        this.CX_CONFIGURATOR_OVERVIEW_MENU
+      )
+    ) {
+      this.configuratorStorefrontUtilsService.syncScroll(
+        this.CX_CONFIGURATOR_OVERVIEW_MENU,
+        element
+      );
+    }
+  }
+
   /**
    * Retrieves the styling for the group levels.
    *
    * @param {number} level - Group level. 1 is top level.
    * @return {string} - corresponding style classes
    */
   getGroupLevelStyleClasses(level: number): string {
-    return 'cx-menu-group' + ' groupLevel' + level;
+    return this.CX_MENU_GROUP + ' groupLevel' + level;
   }
 
   /**
    * Navigates to group in OV form
    *
+   * @param {string} prefix - Prefix (reflects the parent groups in the hierarchy)
    * @param {string} id - Group id
    */
   navigateToGroup(prefix: string, id: string): void {
     const ovGroupId = this.configuratorStorefrontUtilsService.createOvGroupId(
       prefix,
       id
     );
+
     this.configuratorStorefrontUtilsService.scrollToConfigurationElement(
-      '#' + ovGroupId
+      '#' + ovGroupId + ' h2'
```

Comment:
Why that change? Do we need to document it?

Reply:
I don't think that it should be documented. 
The reason of this change is that it will be scrolled to the corresponding group header and not to the group container.

Reply:
Done!